### PR TITLE
rec: add rec_control reload-yaml, document which and how settings can be modified runtime 

### LINKED ISSUE #14870: rec: better docs for runtime changable settings
Some Recursor settings can be changed at runtime via a specific `rec_control command` or `rec_control reload-lua-config`, or API call. We need to systematically document those in the generated documentation. This is important now that _some_ of the YAML settings (that correspond to the old Lua config) and a few new settings are runtime changeable. These can be changed at runtime by `rec_control reload-lua-config`, but that's not documented well. 

A better man page entry and an alias for `rec_control reload-yaml` is needed. The name `reload-lua-config` suggests that it only works for Lua config, while it does work for the YAML settings that correspond to Lua settings.

### PR DESCRIPTION
**Pull request by omoerbeek**
Fixes #14870

Also add ``rec_control reload-yaml`` as an alias for ``reload-lua-config``.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [ ] tested this code
- [X] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/docs/manpages/rec_control.1.rst [hunk 1]
```diff
@@ -211,7 +211,11 @@ reload-lua-config [*FILENAME*]
     executed, any settings changed at runtime that are not modified in this
     file, will still be active. The effects of reloading do not always take
     place immediately, as some subsystems reload and replace configuration
-    in an asynchronous way.
+    in an asynchronous way. IF YAML settings are used this command will
+    reload the runtime settable parts of the YAML settings
```
Comment (Habbie) [inline pdns/recursordist/docs/manpages/rec_control.1.rst:215]:
```suggestion
    in an asynchronous way. If YAML settings are used this command will
    reload the runtime settable parts of the YAML settings.
```

File: pdns/recursordist/rec_channel_rec.cc [hunk 2]
```diff
@@ -1992,7 +1996,7 @@ static RecursorControlChannel::Answer luaconfig(T begin, T end)
 {
   if (begin != end) {
     if (g_luaSettingsInYAML) {
-      return {1, "Unable to reload Lua script from '" + ::arg()["lua-config-file"] + " as there is no active Lua configuration\n"};
+      return {1, "Unable to reload Lua script from '" + *begin + " as there is no active Lua configuration\n"};
```
Comment (Habbie) [inline pdns/recursordist/rec_channel_rec.cc:1999]:
```suggestion
      return {1, "Unable to reload Lua script from '" + *begin + "' as there is no active Lua configuration\n"};
```

File: pdns/recursordist/settings/table.py [hunk 3]
```diff
@@ -40,6 +40,7 @@
 To use this, DNSSEC processing or validation must be enabled by setting :ref:`setting-dnssec` to ``process``, ``log-fail`` or ``validate``.
  ''',
         'versionadded': '4.5.0',
+        'runtime': 'set-max-aggr-nsec-cache-size',
```
Comment (Habbie) [inline pdns/recursordist/settings/table.py:43]: some of the others have `[]`, is there a difference?

Comment (omoerbeek) [reply pdns/recursordist/settings/table.py:43]: a single item gets turned into a list by `generate.py` which takes care of proper formatting and quoting

Comment (Habbie) [reply pdns/recursordist/settings/table.py:43]: ok :)

**Pull request merged**